lib/flakes.nix

```diff
@@ -19,4 +19,24 @@ rec {
     in
     self;
 
+  /* Generates an attrset for the given list of system strings and instantiates
+    nixpkgs from a given flake. The system architecture string for each set is
+    available via the 'system' argument. The instantiated nixpkgs set is available
+    via the 'pkgs' argument.
+
+    Type: forSystems :: [String] -> Flake -> ( String -> AttrSet -> t ) -> AttrSet
+
+    Example:
+      nixpkgs.lib.forSystems
+        [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ]
+        (builtins.getFlake "github:nixos/nixpkgs")
```

Comment:
If you have lib, you probably already have nixpkgs.  Why make nixpkgs be passed in again, when this could just be `pkgs.forSystems`?

But also, at that point I'm not sure it's meaningfully more convenient than genAttrs on its own?

Reply:
I was wrong — nixpkgs needs to be re-evaluated for each system.